python/cudf/cudf/tests/test_parquet.py

```diff
@@ -4341,3 +4342,56 @@ def test_parquet_reader_mismatched_nullability_structs(tmpdir):
         cudf.read_parquet([buf2, buf1]),
         cudf.concat([df2, df1]).reset_index(drop=True),
     )
+
+
+@pytest.mark.parametrize(
```

Comment:
Filtering using StatsAST (no bloom filters here) and BloomFilters (no stats here) yields the same data frames. Though, there is no way other than manual prints/debugging to tell if filtering is yielding correct results. I verified the filtered row groups manually for these cases but we should implement some way to measure this (in another PR perhaps).

Mixing StatsAST and Bloom Filters is unpredictable especially for numeric columns as stats filter numeric cols really well.

Reply:
Thanks for the discussion here. Maybe we can add a trivial check that the ASTs are actually filtering -- that is, the number of rows with no predicate is larger than the number of rows with a predicate (stats or bloom filter). That way we know there is some reduction going on and not just passthrough.

Reply:
Makes sense, though we are already `assert`ing on the expected output table lengths with (either) filtration at L4394-4397. 🤔